src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -60,19 +70,56 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 _pendingProgress = new PendingProgress();
             }
 
-            _pendingProgress.Update(sourceId, record);
```

Comment:
I am trying to avoid this strongly because otherwise we will lose half of optimization effects. 😕 
May we still be able to avoid this?

1. If there is update the percentage, we can safely ignore this loss.
2. If header is updated as in [web cmdlets](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs#L286), then we can safely ignore this loss.
3. If there is an update to a parent progress activity and next a change to a child progress activity, then we have a _problem_ as you pointed out. And I don't see how we can solve it in this place.

<Update>
Strangely when I tested it first, it ate at up to 40%
The current test showed around 5%. Perhaps we have become faster for some other reason.
So I totally agree with your proposal.